### PR TITLE
fix: improve error message for plugin

### LINKED ISSUE #404: Plugin error messages not straightforward for users
### What is not working as expected?

As a user of Notation CLI, the following command:
`notation sign --id $Id --plugin $my_plugin $my_image`

could return the following error:
`Error: describe-key command failed: VALIDATION_ERROR: {plugin specific error details}`

This error message is not very user friendly:
1. What does `describe-key command` stand for?
2. What is been validated here that caused the `VALIDATION_ERROR`?

### What did you expect to happen?

Move the details into verbose/debug logs and a more user-friendly error message might be:
`Error: invalid plugin request: {plugin specific error details}`

### How can we reproduce it?

`notation sign --id $Id --plugin $my_plugin $my_image` with a misconfigured `id`.

The related error message is generated in `notation-go/plugin/plugin.go`.

### Describe your environment

applies to all env

### What is the version of your notation-go Library?

notation-go v1.1.0 with notation CLI v1.0.0+

### PR DESCRIPTION
**Pull request by JeyJeyGao**
Fix:
- Simplified the error message for plugin errors.
- Logged the details of the failing plugin command instead of printing them out in the CLI output, as users may not understand what the plugin command is, which can mislead them about the important part of the error message.
- Logged the error code instead of printing it out in the CLI output to simplify the error message.

Previous example:
```
notation sign xxx.azurecr.io/hello-app:v1 --plugin azure-kv --id https://xxx.vault.azure.net/certificates --plugin-config credential_type=azurecli 
Warning: Always sign the artifact using digest(@sha256:...) rather than a tag(:v1) because tags are mutable and a tag reference can point to a different artifact than the one signed.
Error: failed to execute the describe-key command for azure-kv: VALIDATION_ERROR: Invalid input passed to "--id". Please follow this format to input the ID "https://<vault-name>.vault.azure.net/certificates/<certificate-name>/[certificate-version]"
```

Current example:
```
notation sign xxx.azurecr.io/hello-app:v1 --plugin azure-kv --id https://xxx.vault.azure.net/certificates --plugin-config credential_type=azurecli 
Warning: Always sign the artifact using digest(@sha256:...) rather than a tag(:v1) because tags are mutable and a tag reference can point to a different artifact than the one signed.
Error: failed to sign with the plugin azure-kv: Invalid input passed to "--id". Please follow this format to input the ID "https://<vault-name>.vault.azure.net/certificates/<certificate-name>/[certificate-version]"
```
Resolves #404 
Signed-off-by: Junjie Gao <junjiegao@microsoft.com>

### CODE REVIEW

File: plugin/plugin.go [hunk 1]
```diff
@@ -173,20 +173,21 @@ func run(ctx context.Context, pluginName string, pluginPath string, req plugin.R
...
+			logger.Errorf("failed to execute the %s command for plugin %s: %s", req.Command(), pluginName, err)
 			return &PluginExecutableFileError{
-				Msg:        fmt.Sprintf("failed to execute the %s command for plugin %s", req.Command(), pluginName),
 				InnerError: err,
 			}
 		} else {
 			var re proto.RequestError
 			jsonErr := json.Unmarshal(stderr, &re)
 			if jsonErr != nil {
+				logger.Errorf("failed to execute the %s command for plugin %s: %s", req.Command(), pluginName, strings.TrimSuffix(string(stderr), "\n"))
```
Comment (Two-Hearts) [inline plugin/plugin.go:184]: For `logger.Errorf`, I think the `\n` is redundant?

Comment (JeyJeyGao) [reply plugin/plugin.go:184]: > strings.TrimSuffix(string(stderr), "\n"

This is for removing the new line in the `stderr`, which is useful.

File: plugin/plugin.go [hunk 2]
```diff
@@ -173,20 +173,21 @@ func run(ctx context.Context, pluginName string, pluginPath string, req plugin.R
...
 				return &PluginMalformedError{
-					Msg:        fmt.Sprintf("failed to execute the %s command for plugin %s: %s", req.Command(), pluginName, strings.TrimSuffix(string(stderr), "\n")),
 					InnerError: jsonErr,
 				}
 			}
-			return fmt.Errorf("failed to execute the %s command for plugin %s: %w", req.Command(), pluginName, re)
+			logger.Errorf("failed to execute the %s command for plugin %s: %s:%w", req.Command(), pluginName, re.Code, re)
```
[the remark this answers is not in the document]
Comment (JeyJeyGao) [reply plugin/plugin.go]: Updated.

**Review by Two-Hearts — APPROVED** [review]
PR needs rebase. Otherwise, LGTM.

**Review by priteshbandi — APPROVED** [review]
LGTM

**Pull request merged**